iothub/device/samples/readme.md

```diff
@@ -2,63 +2,69 @@
 This folder contains simple samples showing how to use the various features of Microsoft Azure IoT Hub service, from a device running C# code.
 
 ### [Device samples][device-samples]
-* [Message Sample][d-message-sample]
-* [Method Sample][d-method-sample]
-* [Twin Sample][d-twin-sample]
-* [File Upload Sample][d-file-upload-sample]
-* [Keys Rollover Sample][d-keys-rollover-sample]
-* [Import Export Devices Sample][d-import-export-devices-sample]
-* [Plug And Play Device Sample][d-pnp-sample]
-* [Xamarin Sample][d-xamarin-sample]
+
+- [Reconnection sample][d-message-sample]
+    - This sample illustrates how to write a device application to handle connection issues, connection-related exceptions, and how to manage the lifetime of the `DeviceClient`
+    - Includes sending messages and symmetric key failover
+- [Method sample][d-method-sample]
+- [Twin sample][d-twin-sample]
+- [File upload sample][d-file-upload-sample]
+- [Import/export devices sample][d-import-export-devices-sample]
+- [Plug and Play device samples][d-pnp-sample]
+- [Xamarin sample][d-xamarin-sample]
```

Comment:
there are 2 other samples that are worth adding to this list:
- message receive sample: https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/master/iot-hub/Samples/device/MessageReceiveSample
- using CA-signed x509 device sample: https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/master/iot-hub/Samples/device/X509DeviceCertWithChainSample

Since it is difficult keeping the sourcecode and doc in sync when they are in different locations, how about we move this doc to our samples repo and add a note in the sourcecode repo that informs people about the samples "master" location?

Reply:
Thanks, added.

I'm going to fix the links for now, and we can work out a better strategy with our samples or readme work.